pkg/api/handlers/compat/images.go

```diff
@@ -138,7 +138,7 @@ func CommitContainer(w http.ResponseWriter, r *http.Request) {
 	options.Message = query.Comment
 	options.Author = query.Author
 	options.Pause = query.Pause
-	options.Changes = strings.Fields(query.Changes)
+	options.Changes = strings.Split(query.Changes, ",")
```

Comment:
Test failure suggests that this is not doing the right thing when the `Changes` field is empty. I'm juggling too much this morning to delve deeper into it.

Reply:
My first approach would be to look into `nil` vs `[]` (empty array) vs `""` (empty string), see how `strings.Fields()` differs from `strings.Split()`. HTH.

Reply:
try `s/Println/Printf("%q\n")/`:
```
[]
[""]
```

Reply:
Yikes.